modules/product/src/main/java/com/opengamma/strata/product/credit/type/DatesCdsTemplate.java

```diff
@@ -0,0 +1,447 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.credit.type;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.AdjustablePayment;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.common.BuySell;
+import com.opengamma.strata.product.credit.CdsTrade;
+
+/**
+ * A template for creating credit default swap trades.
+ * <p>
+ * This defines almost all the data necessary to create a credit default swap {@link CdsTrade}.
+ * The start and end of the trade are specified by {@code LocalDate}.
+ * Use {@link TenorCdsTemplate} for standard CDS trades.
+ * <p>
+ * The legal entity ID, trade date, notional and fixed rate are required to complete the template and create the trade.
+ * As such, it is often possible to get a market quote for a trade based on the template.
+ * <p>
+ * A CDS is quoted in points upfront, par spread, or quoted spread. 
+ * For the latter two cases, the market quotes are passed as the fixed rate.
+ */
+@BeanDefinition(builderScope = "private")
+public final class DatesCdsTemplate
+    implements CdsTemplate, ImmutableBean, Serializable {
+
+  /**
+   * The start date.
+   * <p>
+   * The start date of the underling CDS product.
+   * This date can be modified following the rule in {@code convention}. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate startDate;
+  /**
+   * The end date.
+   * <p>
+   * The end date of the underling CDS product.
+   * This date can be modified following the rule in {@code convention}. 
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate endDate;
```

Comment:
A "template" is supposed to be able to create the start/end date from the trade date, so this seems odd.